st2api/tests/unit/controllers/v1/test_kvps.py

```diff
@@ -296,6 +307,146 @@ def test_get_all_all_scope(self):
+            permission_types=[PermissionType.KEY_VALUE_PAIR_DELETE],
+        )
+        grant_5_db = PermissionGrant.add_or_update(grant_5_db)
+
+        # User1
+        role_db = RoleDB(
+            name="custom_role_system_role1_grant",
+            permission_grants=[
+                str(grant_db.id),
+                str(grant_1_db.id),
+                str(grant_2_db.id),
+            ],
+        )
+        role_db = Role.add_or_update(role_db)
+
+        role_assignment_db = UserRoleAssignmentDB(
+            user=user_1_db.name,
+            role=role_db.name,
+            source="assignments/%s.yaml" % user_1_db.name,
+        )
+        UserRoleAssignment.add_or_update(role_assignment_db)
+
+        # User2
+        role_db = RoleDB(
+            name="custom_role_system_role2_grant",
+            permission_grants=[
+                str(grant_3_db.id),
+                str(grant_4_db.id),
+                str(grant_5_db.id),
+            ],
+        )
+        role_db = Role.add_or_update(role_db)
+
+        role_assignment_db = UserRoleAssignmentDB(
+            user=user_2_db.name,
+            role=role_db.name,
+            source="assignments/%s.yaml" % user_2_db.name,
+        )
+        UserRoleAssignment.add_or_update(role_assignment_db)
+
+        mock_system_scoped_kvps.return_value = ["system1", "key4", "key9", "key27"]
+        self.use_user(user_1_db)
+        put_resp_1 = self.__do_put(
+            "system1", {"name": "system1", "value": "val2", "scope": "st2kv.system"}
+        )
+        self.assertEqual(put_resp_1.status_int, 200)
+        put_resp_2 = self.__do_put(
+            "key4", {"name": "key4", "value": "val4", "scope": "st2kv.system"}
+        )
+        self.assertEqual(put_resp_2.status_int, 200)
+        resp = self.app.get("/v1/keys?scope=system")
+        result = sorted(resp.json, key=lambda k: k["name"])
+
+        # asserting the system scope kvps in the response
+        self.assertEqual(result[0]["name"], "key4")
+        self.assertEqual(result[0]["scope"], "st2kv.system")
+        self.assertEqual(result[0]["value"], "val4")
+        self.assertEqual(result[1]["name"], "system1")
+        self.assertEqual(result[1]["scope"], "st2kv.system")
+        self.assertEqual(result[1]["value"], "val2")
+
+        self.__do_delete(self.__get_kvp_id(put_resp_1))
+        self.__do_delete(self.__get_kvp_id(put_resp_2))
+
+        self.use_user(user_2_db)
+        put_resp_3 = self.__do_put(
+            "key9", {"name": "key9", "value": "val9", "scope": "st2kv.system"}
+        )
+        self.assertEqual(put_resp_3.status_int, 200)
+        put_resp_4 = self.__do_put(
+            "key27", {"name": "key27", "value": "val27", "scope": "st2kv.system"}
+        )
+        self.assertEqual(put_resp_4.status_int, 200)
+
+        resp = self.app.get("/v1/keys?scope=system")
+        result = sorted(resp.json, key=lambda k: k["name"])
+
+        # asserting the system scope kvps in the response
```

Comment:
The PR looks good to me but I'd add another assert to ensure that kvps the user is *not* allowed to see (`echo`, `new_action`)  aren't in the result.

Reply:
Refactored the unit tests so your comments here should be addressed.